### PR TITLE
Use an interface to avoid exposing rpc more than necessary.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

**Review by havardpe — COMMENTED** [review]
looks good overall, some minor style comments and one inverted logic issue

File: searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp [hunk 1]
```diff
@@ -344,6 +343,77 @@ writeDomainDir(std::lock_guard<std::mutex> &guard,
...
+class RPCDestination : public Session::Destination {
+public:
+    RPCDestination(FRT_Supervisor & supervisor, FNET_Connection * connection)
+        : _supervisor(supervisor), _connection(connection), _ok(true)
+    {
+        _connection->AddRef();
+    }
+    ~RPCDestination() override { _connection->SubRef(); }
+
+    bool
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp [hunk 2]
```diff
@@ -344,6 +343,77 @@ writeDomainDir(std::lock_guard<std::mutex> &guard,
...
+    {
+        FRT_RPCRequest *req = _supervisor.AllocRPCRequest();
+        req->SetMethodName("visitCallback");
+        req->GetParams()->AddString(domain.c_str());
+        req->GetParams()->AddInt32(id);
+        req->GetParams()->AddData(packet.getHandle().c_str(), packet.getHandle().size());
+        return send(req);
+    }
+
+    bool
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp [hunk 3]
```diff
@@ -344,6 +343,77 @@ writeDomainDir(std::lock_guard<std::mutex> &guard,
...
+    {
+        int32_t retval = rpc(req);
+        if ( ! ((retval == RPC::OK) || (retval == FRTE_RPC_CONNECTION)) ) {
+            LOG(error, "Return value != OK(%d) in send for method 'visitCallback'.", retval);
+        }
+        req->SubRef();
+
+        return (retval == RPC::OK);
+    }
+    int32_t
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp [hunk 4]
```diff
@@ -344,6 +343,77 @@ writeDomainDir(std::lock_guard<std::mutex> &guard,
...
+    {
+        FRT_RPCRequest *req = _supervisor.AllocRPCRequest();
+        req->SetMethodName("eofCallback");
+        req->GetParams()->AddString(domain.c_str());
+        req->GetParams()->AddInt32(id);
+        bool retval(send(req));
+        return retval;
+    }
+    bool connected() const override {
+        return (_connection->GetState() != FNET_Connection::FNET_CONNECTED);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/translogserver.cpp]: Fixed

**Pull request merged**